lib/features/category/presentation/widgets/all_items.dart

```diff
@@ -0,0 +1,56 @@
+import 'package:flutter/material.dart';
+import 'package:shopping_app/core/presentation/widgets/products_grid.dart';
+import 'package:shopping_app/features/product_details/presentation/views/product_details_page.dart';
+
+class AllItems extends StatelessWidget {
+  const AllItems({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return GridView(
+      gridDelegate: const SliverGridDelegateWithFixedCrossAxisCount(
+        crossAxisSpacing: 6,
+        mainAxisSpacing: 10,
+        crossAxisCount: 2,
+      ),
+      children: [
+        GestureDetector(
+          onTap: () => Navigator.of(context).push(MaterialPageRoute(
+              builder: (context) => const ProductDetailsPage())),
+          child: const ProductsGrid(
```

Comment:
Add `onTap` directly to `ProductsGrid` and make this `GestureDetector` internal to `ProductsGrid`.

Also, for the love of God, rename this `ProductsGrid` widget and other creation of yours with the name `'Grid'`